### PR TITLE
Pass block by value

### PR DESCRIPTION
**Pull request by kalverra**
This avoids a weird race condition issue I and @anirudhwarrier caught after some long-running tests.

### CODE REVIEW

File: blockchain/transaction_confirmers.go [hunk 1]
```diff
@@ -71,14 +67,12 @@ func (t *TransactionConfirmer) ReceiveBlock(block NodeBlock) error {
...
-		confirmationLog.Int("Current Confirmations", t.confirmations).
-			Msg("Transaction confirmations met")
+	if t.confirmations >= t.minConfirmations {
+		confirmationLog.Int("Current Confirmations", t.confirmations).Msg("Transaction confirmations met")
 		t.complete = true
 		t.doneChan <- struct{}{}
-	} else if t.confirmations <= t.minConfirmations {
-		confirmationLog.Int("Current Confirmations", t.confirmations).
-			Msg("Waiting on minimum confirmations")
+	} else if t.confirmations < t.minConfirmations {
```
Comment (AnieeG) [inline blockchain/transaction_confirmers.go:74]:
```suggestion
	} else {
```

File: blockchain/transaction_confirmers.go [hunk 2]
```diff
@@ -49,11 +50,6 @@ func NewTransactionConfirmer(client EVMClient, tx *types.Transaction, minConfirm
 // ReceiveBlock the implementation of the HeaderEventSubscription that receives each block and checks
 // tx confirmation
 func (t *TransactionConfirmer) ReceiveBlock(block NodeBlock) error {
-	if block.Block == nil {
-		// Strange case that happens in some EVM testnets
-		log.Debug().Msg("Received nil block")
-		return nil
-	}
```
Comment (AnieeG) [inline blockchain/transaction_confirmers.go:56]: not valid anymore?

File: .nancy-ignore [hunk 3]
```diff
@@ -1,4 +1,5 @@
 CVE-2018-1714 # old helm issue that has all the tickets closed so not much we can do about it
 CVE-2022-23328 # Geth is able to be used for DoS attacks on different nodes. Not something we particularly care about for our tests.
 sonatype-2021-0076 # More weird geth vulnerabilities we don't care about.
-CVE-2022-31030  # This is associated to the dependency 'containerd' and also warns of a DoS attack.g
\ No newline at end of file
+CVE-2022-31030  # This is associated to the dependency 'containerd' and also warns of a DoS attack.
+CVE-2022-37450 # Geth bug that only affects nodes used for mining
```
Comment (AnieeG) [inline .nancy-ignore:5]: would it affect the reorg tests anyhow?

File: blockchain/transaction_confirmers.go [hunk 4]
```diff
@@ -102,22 +96,71 @@ func (t *TransactionConfirmer) Complete() bool {
...
 
-// ReceiveBlock is a no-op
-func (i *InstantConfirmations) ReceiveBlock(block NodeBlock) error {
-	return nil
+func NewL2TxConfirmer(client EVMClient, txHash common.Hash) *L2TxConfirmer {
+	ctx, ctxCancel := context.WithTimeout(context.Background(), client.GetNetworkConfig().Timeout)
+	return &L2TxConfirmer{
+		client:   client,
+		txHash:   txHash,
+		complete: false,
```
Comment (AnieeG) [inline blockchain/transaction_confirmers.go:113]: redundant assignment

File: blockchain/transaction_confirmers.go [hunk 5]
```diff
@@ -102,22 +96,71 @@ func (t *TransactionConfirmer) Complete() bool {
...
-func (i *InstantConfirmations) Wait() error {
+// ReceiveBlock does a quick check on if the tx is confirmed already
+func (l *L2TxConfirmer) ReceiveBlock(block NodeBlock) error {
+	confirmed, err := l.client.IsTxConfirmed(l.txHash)
+	if err != nil {
+		return err
+	}
+	if confirmed {
+		l.complete = true
+	}
```
Comment (AnieeG) [inline blockchain/transaction_confirmers.go:127]:
```suggestion
	l.complete = confirmed
```

**Review by AnieeG — APPROVED** [review]
just few minor questions.

File: .nancy-ignore (hunk 3 above)
Comment (kalverra) [reply .nancy-ignore:5]: We would need to explicitly try and exploit it. And even if we did 🤷

File: blockchain/transaction_confirmers.go (hunk 2 above)
Comment (kalverra) [reply blockchain/transaction_confirmers.go:56]: No, we pass by values instead of pointer now, can't check for nil if we wanted it.

**Review by tateexon — APPROVED** [review]

**Pull request merged**